### PR TITLE
add Compat.Sys.BINDIR

### PR DESCRIPTION
**Pull request by rdeits**
Ref https://github.com/JuliaLang/julia/pull/25102

### CODE REVIEW

File: src/Compat.jl [hunk 1]
```diff
@@ -584,6 +584,10 @@ module Sys
     else
         import Base.Sys: isapple, isbsd, islinux, isunix, iswindows
     end
+
+    @static if VERSION < v"0.7.0-DEV.3073"
+        const BINDIR = JULIA_HOME
+    end
```
[the remark this answers is not in the document]
Comment (rdeits) [reply src/Compat.jl]: thanks! Fixed, and added a test.

**Comment by nalimilan** [comment]
Merge?

**Comment by s2maki** [comment]
This change totally disrupts the ability to do static compilation.  Normally, JULIA_HOME is undefined during static compile because to define it will bake it into the image, which conflicts with the normal initialization when using the new system image.

**Comment by rdeits** [comment]
Should we move the definition into `__init__` ?

**Comment by s2maki** [comment]
I believe it is already in `__init__`.  The problem is that `__init__` is called during static compilation.

It would be better to at least not fail if it doesn't exist.

**Pull request merged**